chromium_src/content/public/browser/tld_ephemeral_storage.h

```diff
@@ -0,0 +1,58 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CONTENT_PUBLIC_BROWSER_TLD_EPHEMERAL_STORAGE_H_
+#define BRAVE_CHROMIUM_SRC_CONTENT_PUBLIC_BROWSER_TLD_EPHEMERAL_STORAGE_H_
+
+#include <string>
+#include <utility>
+#include "base/memory/ref_counted.h"
+#include "base/memory/weak_ptr.h"
+#include "content/common/content_export.h"
+
+namespace content {
+
+class BrowserContext;
+class SessionStorageNamespace;
+class StoragePartition;
+class StoragePartition;
+class WebContents;
+
+// TLD storage is keyed by the BrowserContext (profile) and the TLD-specific
+// security domain.
+using TLDEphemeralStorageKey = std::pair<content::BrowserContext*, std::string>;
+
+// This class is responsible for managing the lifetime of ephemeral storage
+// cookies. Each instance is shared by each top-level frame with the same
+// TLDEphemeralStorageKey. When the last top-level frame holding a reference
+// is destroyed or navigates to a new storage domain, storage will be
+// cleared.
+//
+// TODO(mrobinson): Have this class also manage ephemeral local storage and
+// take care of handing out new instances of session storage.
+class CONTENT_EXPORT TLDEphemeralStorage
```

Comment:
I'm not sure about future plans for this class, but for now it doesn't look like a "storage", rather something like `TLDEphemeralLifetime`

Reply:
I haven't changed this yet, because perhaps it doesn't make sense to rename this class and then rename it back in a followup change. I'm happy to keep discussing it though.

Reply:
But even with encapsulated DOMStorage - would it still be "storage"? For now we basically only keep namespaces and manage their lifetime in the tab helper, instead of "storing" something

Reply:
Okay. I will change the name.